lib/render.test.js

```diff
@@ -227,12 +227,36 @@ describe(_.startCase(filename), function () {
 
     it('redirects if uri matches another uri', function () {
       const resultRef = 'domain.com/_uris/zxccv',
-        mockRes = getMockRes();
+        mockRes = getMockRes(),
+        requestObj = {
+          _parsedUrl: {
+            search: ''
+          }
+        };
 
       // when we find the new reference
       db.get.returns(bluebird.resolve('domain.com/_uris/' + new Buffer(resultRef).toString('base64')));
 
-      return fn('/_uris/asdf', {}, mockRes).then(function () {
+      return fn('/_uris/asdf', requestObj, mockRes).then(function () {
+        expect(mockRes.redirect.callCount).to.equal(1);
+      });
+    });
+
+    it('maintains querystring on redirect', function () {
+      const resultRef = 'domain.com/_uris/zxccv',
+        mockRes = getMockRes(),
+        requestObj = {
+          _parsedUrl: {
+            search: '?query=string'
+          }
+        };
+
+      // console.log("\n\n\n\n MOCK REQ\n\n\n\n", mockReq);
+
+      // when we find the new reference
+      db.get.returns(bluebird.resolve('domain.com/_uris/' + new Buffer(resultRef).toString('base64')));
+
+      return fn('/_uris/asdf', requestObj, mockRes).then(function () {
         expect(mockRes.redirect.callCount).to.equal(1);
```

Comment:
Even though the test passes we're not actually testing if the `res.redirect` method was called with a query string attached to the url. 

I would suggest testing `mockRes.redirect` using `calledWith` to make sure the query param is present